query/src/main/scala/filodb/query/LogicalPlan.scala

```diff
@@ -444,7 +444,7 @@ case class ScalarVectorBinaryOperation(operator: BinaryOperator,
                                        scalarArg: ScalarPlan,
                                        vector: PeriodicSeriesPlan,
                                        scalarIsLhs: Boolean) extends PeriodicSeriesPlan with NonLeafLogicalPlan {
-  override def children: Seq[LogicalPlan] = Seq(vector)
+  override def children: Seq[LogicalPlan] = Seq(vector, scalarArg)
```

Comment:
A `LongTimeRangePlanner` test failed because it (indirectly) expects `getOffsetMillis` to return a `0` for `scalarArg`; the new `getOffsetMillis` uses `children` to walk the tree, and the old used `findLeafLogicalPlans`. I only call it out here because (a) this change would otherwise look random, and (b) it looks like the new plan (i.e. the one given when this extra child isn't included) is a step in the right direction.

Link (with relevant `TODO`):
https://github.com/filodb/FiloDB/blob/1c995cbd7e912cb1a391117069b753d0a5efcf9f/coordinator/src/test/scala/filodb.coordinator/queryplanner/LongTimeRangePlannerSpec.scala#L1405

> // TODO: Though not bad, the plan can be optimized since one operator of this Binary operator is scalar,
> //  entire join should have been pushed down despite offset being present

Query:
```
sum(foo{job="app"} offset 4d) + 10
```

Old plan:
```
        T~ScalarOperationMapper(operator=ADD, scalarOnLhs=false)
        |-FA1~StaticFuncArgs(10.0,RangeParams(1634086130,60,1634777210))
        |-E~StitchRvsExec() on InProcessPlanDispatcher(filodb.core.query.EmptyQueryConfig$@41b13f3d)
        |--T~AggregatePresenter(aggrOp=Sum, aggrParams=List(), rangeParams=RangeParams(1634518490,60,1634777210))
        |---E~LocalPartitionReduceAggregateExec(aggrOp=Sum, aggrParams=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],raw)
        |----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
        |-----T~PeriodicSamplesMapper(start=1634518490000, step=60000, end=1634777210000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
        |------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=9, chunkMethod=TimeRangeChunkScan(1634172590000,1634431610000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],raw)
        |----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
        |-----T~PeriodicSamplesMapper(start=1634518490000, step=60000, end=1634777210000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
        |------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=25, chunkMethod=TimeRangeChunkScan(1634172590000,1634431610000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],raw)
        |--T~AggregatePresenter(aggrOp=Sum, aggrParams=List(), rangeParams=RangeParams(1634086130,60,1634518430))
        |---E~LocalPartitionReduceAggregateExec(aggrOp=Sum, aggrParams=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],downsample)
        |----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
        |-----T~PeriodicSamplesMapper(start=1634086130000, step=60000, end=1634518430000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
        |------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=9, chunkMethod=TimeRangeChunkScan(1633740230000,1634172830000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],downsample)
        |----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
        |-----T~PeriodicSamplesMapper(start=1634086130000, step=60000, end=1634518430000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
        |------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=25, chunkMethod=TimeRangeChunkScan(1633740230000,1634172830000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#-381101616],downsample)
```

Plan becomes:
```
E~StitchRvsExec() on InProcessPlanDispatcher(QueryConfig(10 seconds,300000,1,50,antlr,true,true,None,None,false,false,true))
-T~ScalarOperationMapper(operator=ADD, scalarOnLhs=false)
--FA1~StaticFuncArgs(10.0,RangeParams(1634086130,60,1634777210))
--T~AggregatePresenter(aggrOp=Sum, aggrParams=List(), rangeParams=RangeParams(1634518490,60,1634777210))
---E~LocalPartitionReduceAggregateExec(aggrOp=Sum, aggrParams=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],raw)
----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
-----T~PeriodicSamplesMapper(start=1634518490000, step=60000, end=1634777210000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=9, chunkMethod=TimeRangeChunkScan(1634172590000,1634431610000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],raw)
----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
-----T~PeriodicSamplesMapper(start=1634518490000, step=60000, end=1634777210000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=25, chunkMethod=TimeRangeChunkScan(1634172590000,1634431610000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],raw)
-T~ScalarOperationMapper(operator=ADD, scalarOnLhs=false)
--FA1~StaticFuncArgs(10.0,RangeParams(1634086130,60,1634777210))
--T~AggregatePresenter(aggrOp=Sum, aggrParams=List(), rangeParams=RangeParams(1634086130,60,1634518430))
---E~LocalPartitionReduceAggregateExec(aggrOp=Sum, aggrParams=List()) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],downsample)
----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
-----T~PeriodicSamplesMapper(start=1634086130000, step=60000, end=1634518430000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=9, chunkMethod=TimeRangeChunkScan(1633740230000,1634172830000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],downsample)
----T~AggregateMapReduce(aggrOp=Sum, aggrParams=List(), without=List(), by=List())
-----T~PeriodicSamplesMapper(start=1634086130000, step=60000, end=1634518430000, window=None, functionId=None, rawSource=true, offsetMs=Some(345600000))
------E~MultiSchemaPartitionsExec(dataset=timeseries, shard=25, chunkMethod=TimeRangeChunkScan(1633740230000,1634172830000), filters=List(ColumnFilter(job,Equals(app)), ColumnFilter(__name__,Equals(foo))), colName=None, schema=None) on ActorPlanDispatcher(Actor[akka://default/system/testProbe-1#1715715232],downsample)
```